src/Connection.php

```diff
@@ -256,32 +256,44 @@ public function query(): QueryBuilder
      */
     public function select($query, $bindings = [], $useReadPdo = true): array
     {
-        return $this->run($query, $bindings, function ($query, $bindings) {
-            if ($this->pretending()) {
-                return [];
-            }
-
-            $generator = $this->getDatabaseContext()
-                ->execute($query, ['parameters' => $this->prepareBindings($bindings)])
-                ->rows();
-
-            return iterator_to_array($generator);
-        });
+        return $this->selectWithOptions($query, $bindings, []);
     }
 
     /**
      * @inheritDoc
      */
     public function cursor($query, $bindings = [], $useReadPdo = true): Generator
     {
-        return $this->run($query, $bindings, function ($query, $bindings) {
-            if ($this->pretending()) {
-                return (static fn() => yield from [])();
-            }
+        return $this->cursorWithOptions($query, $bindings, []);
+    }
 
-            return $this->getDatabaseContext()
-                ->execute($query, ['parameters' => $this->prepareBindings($bindings)])
-                ->rows();
+    /**
+     * @param string $query
+     * @param array<array-key, mixed> $bindings
+     * @param array<string, mixed> $options
+     * @return array<int, array<array-key, mixed>>
+     */
+    public function selectWithOptions(string $query, array $bindings, array $options): array
+    {
+        return $this->run($query, $bindings, function ($query, $bindings) use ($options): array {
+            return !$this->pretending()
+                ? iterator_to_array($this->executeQuery($query, $bindings, $options))
+                : [];
+        });
```

Comment:
```suggestion
        return iterator_to_array($this->cursorWithOptions($query, $bindings, $options));
```
How about using `cursorWithOptions`?

Reply:
iterator_to_array must happen within `run` or else certain error cannot be captured.